api/routes/user.py

```diff
@@ -108,3 +111,68 @@ def get_public_top_experts(
     ).all()
 
     return s.PublicTopExpertsOut(top_experts=create_out_search_users(experts, lang, db))
+
+
+@user_router.post("/register-google-account", status_code=status.HTTP_201_CREATED)
+def register_google_account(
+    auth_data: s.GoogleAuthIn,
+    current_user: m.User = Depends(get_current_user),
+    db: Session = Depends(get_db),
+    responses={
+        status.HTTP_409_CONFLICT: {"description": "This Google account is already in use"},
+    },
+):
+    """Register Google account for user"""
+
+    id_info_res: s.GoogleTokenVerification = id_token.verify_oauth2_token(
+        auth_data.id_token,
+        requests.Request(),
+        CFG.GOOGLE_CLIENT_ID,
+    )
+
+    email = id_info_res.email
+    oauth_id = id_info_res.sub
+
+    google_account = get_user_google_account(email, oauth_id, current_user, db)
+
+    if google_account:
+        raise HTTPException(status_code=status.HTTP_409_CONFLICT, detail="This Google account is already in use")
+
+    new_google_account = m.AuthAccount(
+        user_id=current_user.id, email=email, auth_type=s.AuthType.GOOGLE, oauth_id=oauth_id
+    )
+    db.add(new_google_account)
+    db.commit()
+
+    log(log.INFO, "User [%s] successfully added Google account, email: [%s]", current_user.fullname, email)
+
+
+@user_router.post("/delete-google-account", status_code=status.HTTP_204_NO_CONTENT)

```

Comment:
This endpoint is wrong. It should be `@user_router.delete("{auth_account_id}")`

It should take `auth_account_id` as path parameter,
try to find Auth account with  `auth_account_id` and `current_user.id`
return 404 if he fails to find such account

Then check if found AuthAccount is not of type basic, return 409 otherwise

and then it should mark account as deleted